api/utils/slices.go

```diff
@@ -45,17 +45,38 @@ func JoinStrings[T ~string](elems []T, sep string) T {
 	return T(b.String())
 }
 
-// Deduplicate deduplicates list of strings
-func Deduplicate(in []string) []string {
+// Deduplicate deduplicates list of comparable values.
+func Deduplicate[T comparable](in []T) []T {
```

Comment:
nit: I'm not sure about the performance differences, but you could use `DeduplicateAny()` in the implementation here:
```go
func Deduplicate[T comparable](in []T) []T {
	return DeduplicateAny(in, func(t T, t2 T) bool {
		return t == t2
	})
}
```

Reply:
`Deduplicate[T comparable]` has better performance since it can use a `map[T]bool` for O(1) existence checks, so I think it's better not to generalize all uses to the current `DeduplicateAny` with O(n) existence checks. Let me know if you disagree since the performance is pretty negligible to be honest.

I was considering changing `DeduplicateAny` to `func[T any](in []T, hashFunc func(T) string) []T` so `Deduplicate` could simply call that, but I don't think it's worth the extra complexity right now.